### ISSUE TITLE
Support republishing from an array of ObservableObjects

### ISSUE THREAD

**Issue by forgot** [adam-zethraeus/Republished#3]
@adam-zethraeus 

Given something like the following: 

```
final class DomainModel: ObservableObject {

    @Published private(set) var count = 0

    ...

}



@MainActor
final class ViewModel: ObservableObject {

    @Republished private var models: [DomainModel] // <- Error is here

    ...

}
```

Xcode throws the error `Generic class 'Republished' requires that '[DomainModel]' conform to 'ObservableObject'`

Is there a way to propagate changes to a published Array of ObservableObjects?

**Comment by adam-zethraeus** [comment]
Interesting, desirable, use case!

It doesn't work for the same reason that one can't do this from a View:

```swift
// stick this in an xcode playground if you want to repro
import PlaygroundSupport
import SwiftUI

@available(iOS 16, *)
class TestObject: ObservableObject {
  @Published var dataSource: String = "TEXT"
}

@available(iOS 16, *)
public struct ContentView: View {

  // this is fine
  @StateObject var test = TestObject()

  //  expression failed to parse:
  //  error: MyPlayground.playground:12:4: error: generic struct 'StateObject' 
  //  requires that '[TestObject]' conform to 'ObservableObject'
  @StateObject var test2: [TestObject] = [TestObject(), TestObject()]

    public var body: some View {
      VStack {
        Text("hi")
        Text("this")
        Text("is")
        Text(test2.dataSource.first ?? "")
      }
    }
}

PlaygroundPage.current.setLiveView(ContentView())

```

i.e. `TestObject` is an ObservableObject, but an array of them (`[TestObject]`) is not itself an ObservableObject.


It should be possible to make a version of `@Republished` that had a generic parameter setup and the correct overrides to make this doable — though.

So: **Thumb this comment if you've hit this issue!**
(or push a pr to fix it if you're up for it!)

**Comment by adam-zethraeus** [comment]
@forgot if you're still interested here, could you try targeting the experimental branch in https://github.com/adam-zethraeus/Republished/pull/5 and see if it works well for you?

**Comment by adam-zethraeus** [comment]
implemented in 1.1
